cmplibrary/src/main/java/com/sourcepoint/gdpr_cmplibrary/GDPRConsentLib.java

```diff
@@ -285,26 +284,27 @@ public void onMsgShowOptions(){
     }
 
     private void loadPm() {
-        loadConsentUI(pmUrl());
         isPmOn = true;
+        loadConsentUI(pmUrl());
     }
 
     private void loadConsentUI(String url){
-        mCountDownTimer = getTimer(defaultMessageTimeOut);
         mCountDownTimer.start();
+        // warning: ninja code ahead!
         if(webView == null) {
             webView = buildWebView();
             webView.loadConsentUIFromUrl(url);
+        //post(Runnable) must be called if loadUrl(String) was already called on the same instance
         } else if(!isNative) {
             webView.post(new Runnable() {
                 @Override
                 public void run() {
                     webView.loadConsentUIFromUrl(url);
```

Comment:
@leodoin what if we implemented a method like:
```java
void loadConsentUIFromUrlInAnotherThread(webview, url) {
  webView.post {
    webView.loadConsentUIFromUrl(url);
  }
}
```

And we always used `loadConsentUIFromUrlInAnotherThread` instead? It seems to me that, loading the webview in another thread might actually be the way to go.

Reply:
@andresilveirah , thanks for the clean code feedback. Much appreciated :)